upgrade-tests/src/test/java/io/zeebe/test/UpgradeTest.java

```diff
@@ -194,33 +271,56 @@ public void oldGatewayWithNewBroker() {
     TestUtil.waitUntil(() -> state.hasElementInState(PROCESS_ID, "ELEMENT_COMPLETED"));
   }
 
+  @Ignore("https://github.com/zeebe-io/zeebe/issues/5385")
   @Test
   public void upgradeWithSnapshot() {
-    upgradeZeebe(false);
+    upgradeZeebe(true);
   }
 
   @Test
   public void upgradeWithoutSnapshot() {
-    upgradeZeebe(true);
+    upgradeZeebe(false);
   }
 
-  private void upgradeZeebe(final boolean deleteSnapshot) {
+  private void upgradeZeebe(final boolean withSnapshot) {
     // given
-    state.broker(LAST_VERSION, tmpFolder.getRoot().getPath()).start();
+    state.broker(LAST_VERSION, tmpFolder.getRoot().getPath()).start(true);
     final long wfInstanceKey = testCase.setUp(state.client());
     final long key = testCase.runBefore(state);
 
     // when
-    state.close();
     final File snapshot = new File(tmpFolder.getRoot(), "raft-partition/partitions/1/snapshots/");
 
-    assertThat(snapshot).exists();
-    if (deleteSnapshot) {
-      Files.delete(snapshot);
+    if (withSnapshot) {
```

Comment:
This looks like a temp fix for https://github.com/zeebe-io/zeebe/issues/5385. Then why don't the tests works with snapshot?

Reply:
In some cases, the broker creates no snapshot. It prints the message: `We will skip taking this snapshot, because we haven't processed something yet.`.

Reply:
The snapshot is not created because of a bug in `0.24.0` that is fixed in a later version.